### ISSUE TITLE
Timeout seems does not work properly

### ISSUE THREAD

**Issue by horacehylee** [hymkor/expect#31]
Thanks for this awesome library!

But I have tried doing as suggested  `timeout=1` of setting timeout, but seems it does not work and return the RC after it?

**Comment by hymkor** [comment]
Thank you for reporting.

After I read this issue, I tried to confirm and I found the bug on v0.7.\*.
On the timeout, the expect.exe crashes.

```
$ type foo.lua

timeout = 1
echo("do expect")
expect("hoge")
echo("done expect")

<DESKTOP-LGGUCRA:~/goes/expect>
$ expect.exe foo.lua
expect.lua snapshot-windows-amd64
do expect
runtime error: invalid memory address or nil pointer dereference
stack traceback:
        [G]: in function 'expect'
        foo.lua:4: in main chunk
        [G]: ?
exit status 1
```

Is it same as what you see in your environment and script ?

Would you run [v0.6.2](https://github.com/hymkor/expect/releases/tag/v0.6.2) 
and tell me whether the same problem occurs or not  ?

**Comment by hymkor** [comment]
Fixed on [Release v0.7.1 · hymkor/expect](https://github.com/hymkor/expect/releases/tag/v0.7.1)

**Comment by horacehylee** [comment]
Thank you!
